src/qcodeedit/lib/qeditor.cpp

```diff
@@ -3072,14 +3072,17 @@ void QEditor::selectOccurence(bool backward, bool keepMirrors, bool all)
 void QEditor::setVerticalScrollBarMaximum()
 {
 	if (!m_doc) return;
-	const QSize viewportSize = viewport()->size();
-	int viewportHeight = viewportSize.height();
-	if (flag(VerticalOverScroll))
-		viewportHeight /= 2;
-    const qreal ls = m_doc->getLineSpacing();
+	const qreal ls = m_doc->getLineSpacing();
+	const int totalLines = qRound(m_doc->height() / ls);
+	const int viewLines = qFloor(viewport()->size().height() / ls);

```

Comment:
different from old code ?
And also dangerous to accidentally take integer calculus
(that is why the calculations start with 1. (1.0) to force the compiler to calculate float)

Reply:
To  be consequent ``1. + (m_doc->height() - viewportHeight) / ls`` should
be changed to ``1. + 1. * (m_doc->height() - viewportHeight) / ls`` . Really?
Variable ``ls`` is not an integer so no integer division happens.

Reply:
Okay, I have looked it up, this should be fine.
But I have seen cases, where it makes a difference.